torch/_refs/__init__.py

```diff
@@ -3007,14 +3007,23 @@ def expand(a: Tensor, *shape) -> Tensor:
     for idx, x in enumerate(a.shape):
         offset_idx = idx + offset
         requested_length = shape[offset_idx]
+
+        # Similar to the note in fake_impls.py:infer_size, we check for special paths first:
+        # 1) requested_length == -1 -> size & stride are preserved
+        # 2) x == 1 -> size is expanded, and stride becomes 0
+        # if these aren't known to be true, we assume the general case, where x == requested_length
+        # and the size & stride are preserved, and we turn this condition into an expect_true.
         torch._check(
-            guard_size_oblivious(requested_length == x)
-            or guard_size_oblivious(x == 1)
-            or requested_length == -1,
+            guard_or_false(requested_length == -1)
+            or guard_or_false(x == 1)
+            or requested_length == x,
             lambda: f"expand: attempting to expand a dimension of length {x}!",
         )
-
-        shape_[offset_idx] = requested_length if requested_length != -1 else x
+        if guard_or_false(requested_length == -1):
```

Comment:
I think lets use the gaurd_or_else approach so we make it more clear that we are only adding 
torch._check(requested_length >= 0) on a dde

Reply:
```
  if guard_or_false(requested_length == -1, assert_message = "" ):
       shape_[offset_idx] = x
    else:
      shape_[offset_idx] = requested_length
```